tamr_unify_client/models/dataset/resource.py

```diff
@@ -66,20 +68,60 @@ def update_records(self, records, **json_args):
         :rtype: :py:class:`dict`
         """
 
-        def _stringify_updates(updates):
-            for update in updates:
+        def _stringify_updates(updaters):
+            for update in updaters:
                 yield json.dumps(update, **json_args).encode("utf-8")
 
         return (
             self.client.post(
                 self.api_path + ":updateRecords",
                 headers={"Content-Encoding": "utf-8"},
-                data=_stringify_updates(records),
+                data=_stringify_updates(updates),
             )
             .successful()
             .json()
         )
 
+    def upsert_records(self, records, key_name, **json_args):
```

Comment:
nit: I think `primary_key` might be a clearer name than `key_name` since "primary key" is a Unify datasets concept, where as "key" and "name" are generic. Could be `primary_key_name` if we want to be more explicit.